base/dates/io.jl

```diff
@@ -152,7 +152,7 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Period[]
```

Comment:
@felipenoris actually `extra` is used but not within dates. Extra is used by external modules extending the supported slots. Particularly this is used by TimeZones.jl to add support for parsing time zones ([example](https://github.com/quinnj/TimeZones.jl/blob/master/src/timezones/io.jl)). We should definitely add some comments and probably an testcase that checks for extensibility support.